validator/keymanager/remote-web3signer/client.go

```diff
@@ -93,33 +100,32 @@ func (client *client) Sign(pubKey string, request *SignRequest) (bls.Signature,
 	if err := client.unmarshalResponse(resp.Body, &signResp); err != nil {
 		return nil, err
 	}
-	decoded, err := decodeHex(signResp.Signature)
+	decoded, err := hexutil.Decode(signResp.Signature)
 	if err != nil {
-		return nil, err
+		return nil, errors.Wrap(err, "failed to decode signature")
 	}
 	return bls.SignatureFromBytes(decoded)
 }
 
 // GetPublicKeys is a wrapper method around the web3signer publickeys api (this may be removed in the future or moved to another location due to its usage).
-func (client *client) GetPublicKeys() ([][]byte, error) {
-	const requestPath = "/publicKeys"
-	resp, err := client.doRequest(http.MethodGet, client.BasePath+requestPath, nil)
+func (client *client) GetPublicKeys(url string) ([][48]byte, error) {
```

Comment:
Most methods should take in a context as a first argument `ctx context.Context, url string`, as it is a common pattern in the codebase